deposit_contract.sol

```diff
@@ -133,6 +133,10 @@ contract DepositContract is IDepositContract {
         }
     }
 
+    function supportsInterface(bytes4 interfaceId) external pure returns (bool) {
+        return interfaceId == this.supportsInterface.selector || interfaceId == type(IDepositContract).interfaceId;
```

Comment:
Why not properly inherit from the EIP165 interface and check for its ID?

Reply:
```
        return interfaceId == type(IEIP165).interfaceId || interfaceId == type(IDepositContract).interfaceId;
```

You mean this?

Reply:
yes